src/ripple/rpc/handlers/Ping.cpp

```diff
@@ -39,13 +39,14 @@ doPing(RPC::JsonContext& context)
             break;
         case Role::IDENTIFIED:
             ret[jss::role] = "identified";
-            ret[jss::username] = context.headers.user.to_string();
+            // Can I avoid making expensive copies for std::string?
+            ret[jss::username] = std::string{context.headers.user};
             if (context.headers.forwardedFor.size())
```

Comment:
@scottschurr  I couldn't think of a cheaper solution for this piece of code.
`std::string{context.headers.user.data()}` seems to include garbage trailing data

Reply:
At this level of the source code I'm not seeing `boost::string_view` used here.  Naively, it seems like you can just leave this alone?  Am I missing something?

Remember, there are _lots_ of places that `std::string_view` is not safe to use.  We should not try to replace uses of `std::string` with `std::string_view` unless we're really sure it is safe.  I suggest you not try _any_ replacements of `std::string` with `std::string_view` in this initial pass.

Reply:
I understand. `context.headers.user` and `context.headers.forwardedFor` are of type `boost::string_view` in the current version. Source: `src/ripple/rpc/Context.h:60`

I'm trying to port from this dependence on boost to std.

Reply:
Thanks for the clarification.  I don't think there's currently any way to make this cheaper than what you have done here.  The `Json::Value::operator[]` interface does not (yet) allow assignment from a `std::string_view`.But it could be made `string_view` aware in a future pull request if we chose to do that.

Alternatively, we could update the `rippled` codebase to use `boost::json` instead of `Json::Value`.  I think that would also get us `std::string_view` support.  But that would be a much bigger lift (with potentially more benefit).

Reply:
ok, thanks for the suggestions Scott! 
Yeah, `Json::Value` was written a long time, I'm guessing before boost came up with their json library. It appears to have so much of overlap with implementations from boost!